kafka-rest/src/main/java/io/confluent/kafkarest/KafkaRestConfig.java

```diff
@@ -365,6 +371,43 @@
       "Whether to enable REST Proxy V3 API. Default is true.";
   private static final boolean API_V3_ENABLE_DEFAULT = true;
 
+  public static final String RATE_LIMIT_ENABLE_CONFIG = "rate.limit.enable";
+  private static final String RATE_LIMIT_ENABLE_DOC =
+      "Whether to enable request rate-limiting. Default is false.";
+  private static final boolean RATE_LIMIT_ENABLE_DEFAULT = false;
+
+  public static final String RATE_LIMIT_BACKEND_CONFIG = "rate.limit.backend";
+  private static final String RATE_LIMIT_BACKEND_DOC =
```

Comment:
Why the choice of rate limiting back end?  Are we giving the option to test which is more efficient? If so, we should probably remove the option again once we know which is the preferred limiter.

Reply:
I think having two backends (actually 3, if you include the null rate limiter) significantly increases the complexity of the code too - we need delegates and different factories.  With only one limiter, I think a lot of the layering of the classes could potentially be removed.  

I'm not convinced a null rate limiter is required if we only have one rate limiter option - for each call to an endpoint the code is having to work out which rate limiter to use.  If you have just the one, it's probably more efficient to go "if rate limiting disabled, just return" next to the call to the guava/resilience4j call.

Reply:
I actually removed a whole layer of dependency injection by using a dynamic-feature instead (in the latest commit). That should make it much more efficient (the rate-limiter decision is only made once, at startup, per endpoint). I have two backends because I don't have enough data to decide which one is better. They use different algorithms, and on the tests, behave differently. I think once we merge this PR, it's unlikely we will remove the backend choice (for backwards compatibility). I don't think that's a problem though.